code/LeftAndMain.php

```diff
@@ -773,6 +778,32 @@ protected function init()
         Versioned::set_default_reading_mode(Versioned::get_reading_mode());
     }
 
+    public function afterHandleRequest()
+    {
+        if ($this->response->isError()) {
+            $this->init();
+            $errorCode = $this->response->getStatusCode();
+            $errorType = $this->response->getStatusDescription();
+            $defaultMessage = _t(
+                self::class . '.ErrorMessage',
+                'Sorry, it seems there was a {errorcode} error.',
+                ['errorcode' => $errorCode]
+            );
+            $this->response = HTTPResponse::create($this->render([
+                'Title' => $this->getApplicationName() . ' - ' . $errorType,
+                'Content' => $this->renderWith($this->getTemplatesWithSuffix('_NotFound'), [
```

Comment:
Should rename the new template to `_Error` or something generic as probably not all errors are 'not-found'

Reply:
Ahh yup my bad. I was originally only treating 404 but forgot to change this when I generalised it. Will fix.